server/src/main/java/org/apache/iotdb/db/metadata/MManager.java

```diff
@@ -1241,8 +1241,10 @@ public IMNode getDeviceNodeWithAutoCreate(PartialPath path)
       Template template = node.getUpperTemplate();
 
       for (IMNode child : node.getChildren().values()) {
-        IMeasurementMNode measurementMNode = (IMeasurementMNode) child;
-        res.add(measurementMNode.getSchema());
+        if (child instanceof IMeasurementMNode) {
```

Comment:
It's better use override method child.isMeasurement.
Java instansof violates OO design and program.

Reply:
Thanks for your reply!
I have changed to is measurement function.